service_worker.js

```diff
@@ -249,8 +252,18 @@ async function animateBadges(request, tabId) {
   const delay = 2000;
   // First badge overall perf
   badgeOverallPerf(request.passesAllThresholds, tabId);
+
   // If perf is poor, animate the sequence
   if (request.passesAllThresholds === 'POOR') {
+
+    // However, if user has turned this off, then leave it off.
+    // Note, we don't call animateBadges again so need a new metric event or a
```

Comment:
```suggestion
    // TODO(tunetheweb): we don't call animateBadges again so need a new metric event or a
```
?

Reply:
I don't think this is a TODO so much as an observation.

If you're RED and not animating, and then re-enable animation it won't start animating immediately until a status changes or the page reload. I think that's fair enough, and not worth the complexity to reenable animations immediately. In the same way as the other options HUD, console logging don't activate immediately

WDYT?

Reply:
That's fine, maybe reword the comment slightly then? I just suggested it because it currently sounds like a TODO :)

Adapting your comment here, maybe something like:
```js
    // Note: if optionsNoBadgeAnimation is flipped, it won't start (or stop)
    // animating immediately until a status change or page reload to avoid
    // having to check continually.
```
or whatever sounds good.